effectsize_supplement.Rmd

```diff
@@ -0,0 +1,281 @@
+```
+
+This shows priors on the log-scale mean and priors on the log-scale standard deviation (`sigma`).
+
+First, we'll assume that completion time is something like a pointing task: not reasonably faster than 10ms or slower than 2s (2000ms). On log scale, that is between approximately $\log(10) \approx 2$ and $\log(2000) \approx 8$, so we'll center our prior intercept between these ($(8+2)/2$) and give it a 95% interval that covers them (sd of $(8-2)/4$: $\mathrm{\Normal}(5, 1.5)$.
+
+For differences in log mean, we'll assume that times will not be more than about 100$\times$ difference in either direction: a zero-centered normal prior with standard deviation $\approx log(100)/2 \approx 2.3$: $\mathrm{Normal}(0, 2.3)$.
+
+For standard deviation, we will assume that individual differences are not generally as large: somewhere between 10ms and 1000ms. $\log(1) = 0$ and $\log(1000) \approx 7$, so we'll set a $\mathrm{Normal}(3.5, 1.75)$ prior on the intercept of the standard deviation on log scale.
+
+Finally, for differences in log-scale standard deviation, we'll assume zero-centered with similar magnitude to the intercept: $\mathrm{Normal}(0, 1.75)$  [todo: not sure I've got sensible sd priors here].
+
+These priors can be specified as follows:
+
+```{r}
+log_bayes_priors <- c(
+    prior(normal(5.5, 1.75), class = b, coef = intercept),
+    prior(normal(0, 2.3), class = b, coef = groupB),
+    prior(normal(3.5, 1.75), class = b, coef = intercept, nlpar = sigma),
+    prior(normal(0, 1.75), class = b, coef = groupB, nlpar = sigma)
+  )
+log_bayes_priors
+```
+
+Then we can re-run the model from approach 3a with those priors:
+
+```{r, results = "hide"}
+m_log_bayes_informed <- brm(brmsformula(
+    completion_time_ms ~ group,
+    sigma ~ group
+  ), data = data, family = lognormal, prior = log_bayes_priors)
+```
+
+Similar to approach 2b, we will derive samples of the mean difference, this time from the posterior distribution. We will use these to derive a credible interval (Bayesian analog to a confidence interval) around the mean difference:
+
+```{r}
+log_bayes_informed_result <- m_log_bayes_informed %>%
+  as_data_frame() %>%
+  mutate(
+    mu_A = b_Intercept,
+    sigma_A = exp(b_sigma_Intercept),
+    mean_A = exp(mu_A + sigma_A^2 / 2),
+    
+    mu_B = b_Intercept + b_groupB,
+    sigma_B = exp(b_sigma_Intercept + b_sigma_groupB),
+    mean_B = exp(mu_B + sigma_B^2 / 2),
+    
+    mean_difference = mean_A - mean_B
+  ) %>%
+  mean_qi(mean_difference) %>%
+  mutate(method = "lognormal regression (Bayesian, weakly informed)")
+log_bayes_informed_result
+```
+
+This gives the estimated mean difference between conditions in milliseconds (`mean_difference`), as well as the lower (`mean_difference.lower`) and upper (`mean_difference.upper`) bounds of the 95% quantile credible interval of that ratio.
+
+
+
+### Comparing approaches
+
+All approaches that give estimates for the difference in means give very similar results:
```

Comment:
Would it be useful to show the shape of the posterior distribution from the Bayesian approaches?